### PR TITLE
Add coupon saving to app database and rspec

### PR DESCRIPTION
**Pull request by OleksaRiabukha**
Coupons from the third party API are now can be saved into the app's database upon successful Stripe payment. RSpec tests were added to test this functionality.

### CODE REVIEW

File: app/models/coupon.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+class Coupon < ApplicationRecord
+  belongs_to :user, optional: true
+
+  validates :coupon_number, :initial_amount, presence: true
+  validates :coupon_number, uniqueness: true
+
+  def self.save_coupon(coupon_params, user)
+    coupon_number, initial_amount, for_present = coupon_params.values_at(:coupon_number, :amount, :for_present)
+    coupon = Coupon.new(coupon_number: coupon_number,
```
Comment (BohdanBusko) [inline app/models/coupon.rb]:
```ruby
coupon = Coupon.new(
  coupon_number: coupon_number,
  initial_amount: initial_amount,
  amount_left: initial_amount,
  for_present: for_present,
  user_id: user.id
)
```

File: app/models/coupon.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                        initial_amount: initial_amount,
+                        amount_left: initial_amount,
+                        for_present: for_present,
+                        user_id: user.id)
+
+    if coupon.valid?
```
Comment (BohdanBusko) [inline app/models/coupon.rb]:
```ruby
if !coupon.valid?
  logger.error "Encountered errors: #{coupon.errors.full_messages}"
  throw :abort
  return
end

coupn.save
```

File: spec/models/coupon_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => users.id)
+#
+require 'rails_helper'
+
+RSpec.describe Coupon, type: :model do
+  describe 'validations' do
+    subject { create(:coupon) }
```
Comment (BohdanBusko) [inline spec/models/coupon_spec.rb:27]: enter after

**Review by BohdanBusko — COMMENTED** [review]
User always should be authorized before buying a coupon

**Comment by OleksaRiabukha** [comment]
> User always should be authorized before buying a coupon

**Pull request merged**